Source/WebKit/NetworkProcess/cache/NetworkCache.cpp

```diff
@@ -284,6 +289,12 @@ static StoreDecision makeStoreDecision(const WebCore::ResourceRequest& originalR
             return StoreDecision::NoDueToHTTPStatusCode;
     }
 
+    // Our caching does not properly handle the case where we perform a redirect with fragments.
+    // If we cache localhost/#key=foo we will return the same parameters key=foo
+    // on the next visit to the domain even if the parameters have been updated.
+    if ((response.isRedirection() || response.isRedirected()) && originalRequest.url().hasFragmentIdentifier())
```

Comment:
It might be good to clarify the message to something like:
```
// FIXME: We are not correctly computing the redirected request URL in case original request
// has a fragment identifier and response location URL does not have one. Let's not store it for now.
```